container/kubernetes/onboarding/deviceonboarding.py

```diff
@@ -213,11 +259,13 @@ def create_new_deployment(deploy_instance, deploy_type):
     # TestSensor doesn't have Machine Config so check and store
     # relevant Machine Config data.
     machineconfig_exist = True
-    if len(deploy_instance) > 6:
+    if len(deploy_instance) > 12:
```

Comment:
It might be better to use a [dataclass](https://docs.python.org/3/library/dataclasses.html) or a [named tuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) instead of a list of varying size here. Right now it works, but it feels like it could create hard to find bugs long term. 

I would consider initiating the object in the `process_deploy_devices_gateway` method then. It would become a little more verbose, but less error prone. Again, not a huge problem the way it is.